### DISCUSSION TITLE
3.0.0 beta 5

### DISCUSSION THREAD

**Discussion by zHaytam** [Blazor-Diagrams/Blazor.Diagrams#271] [Announcements]
### Added

- `AdditionalSvg` option to `DiagramCanvas` in order to render any exatra SVG content you want
- `AdditionalHtml` option to `DiagramCanvas` in order to render any exatra SVG content you want
- `DistanceTo` overload method to `Point` that takes x and y
- `MoveAlongLine` method to `Point`
- `FullPath` to `PathGeneratorResult` to represent the full path without cuts
- Fallback router to Orthogonal router
- Margin options to `OrthogonalRouter`
- `radius` option to `StraightPathGenerator` in order to generate rounded bends
- Support for custom vertices
- `AutoSize` option to groups to control whether moving children resizes the group

### Changed

- All routers are now classes instead of functions, they inherit from the new abstract class `Router`
- All path generators are now classes instead of functions, they inherit from the new abstract class `PathGenerator`
- Optimize Orthogonal router by using custom A* (x5 improvement)

### Removed

- `Router` delegate
- `PathGenerator` delegate

<hr /><em>This discussion was created from the release <a href='https://github.com/Blazor-Diagrams/Blazor.Diagrams/releases/tag/3.0.0-beta.5'>3.0.0 beta 5</a>.</em>

**Comment by JeremyGEA** [comment]
Any roadmap on when version 3 will be the official release?
